File: services/education/internal/model/scale_library.go
package model

type ScaleLibraryQuery struct {
	Keyword  string `json:"keyword,omitempty"`
	Category string `json:"category,omitempty"`
	Scenario string `json:"scenario,omitempty"`
	Status   string `json:"status,omitempty"`
	AgeScope string `json:"ageScope,omitempty"`
	Duration string `json:"duration,omitempty"`
}

type ScaleLibraryTextResource struct {
	ID      int64  `json:"id"`
	ScaleID int64  `json:"scaleId"`
	Content string `json:"content"`
	Sort    int    `json:"sort"`
}

type ScaleLibraryItem struct {
	ID                 int64                      `json:"id"`
	Name               string                     `json:"name"`
	Code               string                     `json:"code"`
	Category           string                     `json:"category"`
	Scenario           string                     `json:"scenario"`
	AgeRange           string                     `json:"ageRange"`
	AgeMinMonths       int                        `json:"ageMinMonths"`
	AgeMaxMonths       int                        `json:"ageMaxMonths"`
	Duration           string                     `json:"duration"`
	DurationMinMinutes int                        `json:"durationMinMinutes"`
	DurationMaxMinutes int                        `json:"durationMaxMinutes"`
	CurrentVersion     string                     `json:"currentVersion"`
	ItemCount          int                        `json:"itemCount"`
	DomainCount        int                        `json:"domainCount"`
	InstitutionCount   int                        `json:"institutionCount"`
	MonthUsage         int                        `json:"monthUsage"`
	UsageCount         int                        `json:"usageCount"`
	LatestUse          string                     `json:"latestUse"`
	DataStatus         string                     `json:"dataStatus"`
	Status             string                     `json:"status"`
	StatusText         string                     `json:"statusText"`
	UpdatedAt          string                     `json:"updatedAt"`
	Summary            string                     `json:"summary"`
	PosterURL          string                     `json:"posterUrl"`
	ExecutionEntry     string                     `json:"executionEntry"`
	APIPackage         string                     `json:"apiPackage"`
	References         []ScaleLibraryTextResource `json:"references"`
	Acknowledgements   []ScaleLibraryTextResource `json:"acknowledgements"`
	AuthReserved       bool                       `json:"authReserved"`
	AuthActionEnabled  bool                       `json:"authActionEnabled"`
}

type ScaleLibrarySummary struct {
	Total         int `json:"total"`
	Available     int `json:"available"`
	Unavailable   int `json:"unavailable"`
	MonthUsage    int `json:"monthUsage"`
	UsageCount    int `json:"usageCount"`
	ReservedAuths int `json:"reservedAuths"`
}

type ScaleLibraryFilterOptions struct {
	Categories     []string       `json:"categories"`
	CategoryCounts map[string]int `json:"categoryCounts"`
	Scenarios      []string       `json:"scenarios"`
	Statuses       []string       `json:"statuses"`
	AgeScopes      []string       `json:"ageScopes"`
	Durations      []string       `json:"durations"`
}

type ScaleLibraryVO struct {
	Items         []ScaleLibraryItem        `json:"items"`
	Summary       ScaleLibrarySummary       `json:"summary"`
	FilterOptions ScaleLibraryFilterOptions `json:"filterOptions"`
}

File: institution-admin/src/api/teacher-center/scale-library.ts
import { useGet } from '~/utils/request'

export interface ScaleLibraryTextResource {
  id: number
  scaleId: number
  content: string
  sort: number
}

export type ScaleLibraryStatus = 'available' | 'unavailable' | string

export interface ScaleLibraryItem {
  id: number
  name: string
  code: string
  category: string
  scenario: string
  ageRange: string
  duration: string
  durationMinMinutes: number
  durationMaxMinutes: number
  currentVersion: string
  itemCount: number
  domainCount: number
  institutionCount: number
  monthUsage: number
  usageCount: number
  latestUse: string
  dataStatus: string
  status: ScaleLibraryStatus
  statusText: string
  updatedAt: string
  summary: string
  executionEntry: string
  apiPackage: string
  references: ScaleLibraryTextResource[]
  acknowledgements: ScaleLibraryTextResource[]
  authReserved: boolean
  authActionEnabled: boolean
}

export interface ScaleLibrarySummary {
  total: number
  available: number
  unavailable: number
  monthUsage: number
  usageCount: number
  reservedAuths: number
}

export interface ScaleLibraryFilterOptions {
  categories: string[]
  scenarios: string[]
  statuses: string[]
  ageScopes: string[]
  durations: string[]
}

export interface ScaleLibraryResponse {
  items: ScaleLibraryItem[]
  summary: ScaleLibrarySummary
  filterOptions: ScaleLibraryFilterOptions
}

export interface ScaleLibraryQuery {
  keyword?: string
  category?: string
  scenario?: string
  status?: string
  ageScope?: string
  duration?: string
}

export function getScaleLibraryApi(params?: ScaleLibraryQuery) {
  return useGet<ScaleLibraryResponse, ScaleLibraryQuery>('/api/v1/assessments/scales/library', params)
}
